packages/forklift-console-plugin/src/modules/Providers/views/create/components/VCenterProviderCreateForm.tsx

```diff
@@ -87,12 +81,10 @@ export const VCenterProviderCreateForm: React.FC<VCenterProviderCreateFormProps>
         onChange({
           ...provider,
           spec: {
-            type: provider.spec.type,
-            url: provider.spec.url,
             ...provider?.spec,
             settings: {
               ...(provider?.spec?.settings as object),
-              vddkInitImage: trimmedValue || undefined,
+              vddkInitImage: trimmedValue,
```

Comment:
the idea of this line if to delete  `vddkInitImage` in case it's ''.
are you sure this still works ? what happen if `trimmedValue === ''` ? can `trimmedValue` be '' ?

Reply:
It still works,  since `empty_str = String.trim(empty_str) `

The current behavior sets  all fields to an `empty string` in case the user sets and then cleans the UI field. This is the only way to distinguish between the init state (`undefined`) to other empty states (`''`). For me it works fine.

Reply:
i am missing something, 
when setting a k8s CR (e.g. using `k8sCreate` or `k8sPatch`):
'' means set the value to ''
`undefined` means `delete` field

you should check the result resource that get created in k8s,  the check is: the field get deleted and not just set to '' value, this needs to be verified, and commented on somehow, `undefined` is being explicit that we mean to delete, when we see `trimmedValue` it's not clear, so we need some way to make sure a user ( of the code e.g. dev ) will know that we explicitly want to delete and not set to some value.

Reply:
Good point since this is a special case indeed. I wanted to reserve the "warning" validation state for the   `vddkInitImage` field, but this is a problem since backend won't allow saving such CR.

So let's revert to original code. The effect is that switching endpoint types will reset the validation back to `default` validation state, which seems ok to me since it's only a warning.